File: 006-Get-Middle-Character/main.py
def get_middle(s):
    length = len(s)
    
    if length % 2 !=0:
        value = (length - 1) / 2
        text = s[int(value)]
        
    if length % 2 == 0:
        value = (length) / 2
        text = s[int(value)-1] + s[int(value)]
    
    print(text)
    return text
    
get_middle("textin")

File: 006-Get-Middle-Character/main.cpp
#include <iostream>
#include <string>

using namespace std;

string get_laptop(string s)  {
    int length = s.length();
    string result;

    // for even number of characters in the string.
    if (length % 2 == 0 ) {

        int value = length / 2;
        result = string(1, s[value - 1]) + s[value];

    } else {
        int value = length / 2;
        result = string(1, s[value]);
    }
    return result;
}

int main() {
    cout << "Current state: " << get_laptop("textin") << endl;
    cout << "Current state: " << get_laptop("hello") << endl;
    cout << "Current state: " << get_laptop("hello") << endl;
    return 0;
}